projects/onto-search/src/lib/onto-search-facet/range-facet/range-slider/range-slider.component.ts

```diff
@@ -0,0 +1,156 @@
+import {AfterViewInit, Component, ElementRef, EventEmitter, Input, OnInit, Output, ViewChild} from '@angular/core';
+import {SearchFacetGroupModel} from '../../models/search-facet-group-model';
+import {SearchFacetModel} from '../../models/search-facet-model';
+
+@Component({
+  selector: 'onto-range-slider',
+  templateUrl: './range-slider.component.html',
+  styleUrls: ['./range-slider.component.scss']
+})
+export class RangeSliderComponent implements OnInit, AfterViewInit {
+  @Input()
+  public data: SearchFacetGroupModel;
+
+  @ViewChild('minMaxSlider')
+  public slider: ElementRef;
+  @ViewChild('minSlider')
+  public minSlider: ElementRef;
+  @ViewChild('maxSlider')
+  public maxSlider: ElementRef;
+
+  @Output()
+  public sliderWidth: EventEmitter<any> = new EventEmitter<any>();
+
+  @Output()
+  public selectionChange: EventEmitter<any> = new EventEmitter<any>();
+
+
+  private facetData: SearchFacetModel[];
+  public sum: number;
+
+  private thumbsize = 14;
+  public min;
+  public max;
+  public selectedMin;
+  public selectedMax;
+
+  ngOnInit(): void {
+    this.facetData = JSON.parse(JSON.stringify(this.data.facetGroupData));
+    this.facetData.sort((a, b) => parseInt(a.label) - parseInt(b.label));
+    this.min = this.facetData[0].label;
+    this.max = this.facetData[this.facetData.length-1].label;
+    this.selectedMin = this.min;
+    this.selectedMax = this.max;
+
+    this.sum = this.sumRange();
+    this.updateSelection();
+  }
+
+  public ngAfterViewInit(): void {
+    this.init();
+    this.sliderWidth.emit(this.slider.nativeElement.clientWidth);
```

Comment:
This should emit on every width change not only after init

Reply:
done